Moved Initialisation of General and Divisional templates from GOOrganController to GOOrganModel

oleg68 opened the pull request:
This is a new PR instead of https://github.com/GrandOrgue/grandorgue/pull/1576

In order to eliminating usage of GOOrganController I moved storing and initialisation of GOGeneralCombination GODivisionalCombination from GOOrganController to GOOrganModel.

It is just refactoring. No GO controller should be changed.

After merging https://github.com/GrandOrgue/grandorgue/pull/1580 the capability of saving virtual coupler states are saved correctly with this PR.

larspalo reviewed (changes requested):
With this PR the CombinationStoreNonDisplayedDrawstops suddenly becomes necessary as it won't default to true as before. I suggest that you change it back to default to true instead of being a mandatory line in the odf. There might be more issues, but this is the first I immediately found.

oleg68 commented:
@larspalo 

> With this PR the CombinationStoreNonDisplayedDrawstops suddenly becomes necessary as it won't default to true as before. I suggest that you change it back to default to true instead of being a mandatory line in the odf.

At #1498 the semantics of `CombinationsStoreNonDisplayedDrawstops` was changed totally.

Now it relates to being read-only instead of the visibility. Earlier there were problems with determining visibility, but now there are no more issues with it.

After fixed the problem with storing read-olny elements at https://github.com/GrandOrgue/grandorgue/pull/1582/files#diff-e96a2e520f7cdeb61d46c74147ca677d9f681b3590651d84a5cd82b912d230fe the whole set of rad-only drawstops started spaming in yaml, so I switched it's default to `false`.

I changed this default to `false` because n

Do you know an example of ODF where CombinationsStoreNonDisplayedDrawstops=Y is really necessary? Otherwise I'd leave the new default.

larspalo commented:
The new PR breaks among others Burea Church and Norrfjärden Church sample sets. The thing is that this PR doesn't just change the default value, it makes the line necessary to include in the odf, whereas earlier one could omit it and trust the default value to kick in. I suggest to just revert the default value to true and keep the CombinationStoreNonDisplayedDrawstops as not necessary to have included.

Personally, I'd even say that it's not even necessary to keep CombinationStoreNonDisplayedDrawstops as a parameter in the odf (should be depracated and kept only for legacy reasons) since the displayed property is no longer the deciding factor - the read only property is now taking its place.

Anyway, the important thing is that it should not be needed to include in the odf, since that does indeed change GO behavior.

oleg68 commented:
> I suggest to just revert the default value to true and keep the CombinationStoreNonDisplayedDrawstops as not necessary to have included.

I marked this parameter as not mandatory, but kept the default value as `false`.

larspalo commented:
@oleg68 Yes, that works for me. Only thing left then is that the help should also be updated to state that the default value is false now like so:

```
CombinationsStoreNonDisplayedDrawstops
(boolean, default: false) determines, if the state of invisible objects (on the main panel) is stored in divisionals, generals and the setter. 
```

Also, perhaps a note should be added that this is actually only used for old style panel format .organ files (if even that) since that mechanism has been changed to actually work with how the new panel format does things. I leave that up to you though.

oleg68 commented:
> @oleg68 Yes, that works for me. Only thing left then is that the help should also be updated to state that the default value is false now like so:
> 
> ```
> CombinationsStoreNonDisplayedDrawstops
> (boolean, default: false) determines, if the state of invisible objects (on the main panel) is stored in divisionals, generals and the setter. 
> ```

Fixed.

> Also, perhaps a note should be added that this is actually only used for old style panel format .organ files (if even that) since that mechanism has been changed to actually work with how the new panel format does things.

Won't fix, because `CombinationsStoreNonDisplayedDrawstops` behavior does not depend on the panel format style.
